src/System.Buffers/tests/System.Buffers.Tests.csproj

```diff
@@ -13,14 +13,22 @@
     <Compile Include="$(CommonPath)tests\TestUtilities\System\PlatformDetection.cs" Link="Common\System\PlatformDetection.cs" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(OS)' != 'Windows_NT'">
+    <Compile Include="$(CommonPath)tests\TestUtilities\System\PlatformDetection.Unix.cs" Link="Common\System\PlatformDetection.Unix.cs" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(OS)' == 'Windows_NT'">
+    <Compile Include="$(CommonPath)tests\TestUtilities\System\PlatformDetection.Windows.cs" Link="Common\System\PlatformDetection.Windows.cs" />
+  </ItemGroup>
```

Comment:
I think it would be better to auto-include these compile items for test projects in the repo's Directory.Build.targets file. Including the PlatformDetection.cs compile item above.